kwanj-k/storemanager-v2
# #161617593 Validation bug fixes

### What does this PR do?
Fixes more edge cases on user inputs

#### Description of Task to be Completed?
-- Validate all email edge cases
-- Convert strings to lower case before populating db
-- Add an edit password endpoin
-- Add a delete User endpoint for the store owner

### How should this be manually tested?
* After cloning the repo, git pull the branch bg-validation-#161617593

* Create a virtual enviroment and activate it : :
    --- virtualenv -p python3 venv
   
    --- source venv/bin/activate

* Run the following:
  ``` export FLASK_APP='run.py' ```
  ``` export APP_SETTINGS='development' ```


Also, you need to create a development and test database and connect to it.[Refer here](https://www.digitalocean.com/community/tutorials/how-to-install-and-use-postgresql-on-ubuntu-18-04#creating-a-new-role)

Then export the database URL in the environment:
```
export Dev_URL="dbname=dbname user=username password=password"
export Test_URL="dbname=dbname user=username password=password"
export SECRET="iamasecrettheunsadtheunsulliedtheundead"

```
We need the secret key for generating the tokens

Then Run the tests:
```
pytest --cov-report term-missing --cov=app
```
* All the tests should pass:

### What are the relevant Pivotal Tracker stories?
  #161617593
